### PR TITLE
Added support for custom labels and pipelines

### PR DESCRIPTION
**Pull request by pgarg66**
- A list of labels can be declared as environment variable
- If a PR is tagged with any of these labels, the corresponding
  buildkite pipeline will be triggered

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -445,6 +506,12 @@ async function onGithubPullRequest(payload) {
       }
       await autoMergePullRequest(repoName, prNumber);
     }
+    for (let index = 0; index < customLabels.length; ++index) {
```
Comment (mvines) [inline index.js]: Let's merge this block and 487-492 into a function, it's the exact same code I think

File: index.js [hunk 2]
```diff
@@ -70,6 +70,12 @@ const envconst = {
      automatically granted CI access
    */
   CI_USER_WHITELIST: '', // comma separated, no spaces
+
+  /*
+     Labels that are specific to instance of CI gate.
+     These labels are passed on to buildkite.
+   */
+  CUSTOM_LABELS: 'testnet,', // comma separated, no spaces
```
Comment (mvines) [inline index.js]: `CUSTOM_LABELS` should be the empty string for the default configuration

Comment (mvines) [reply index.js]: How about `CUSTOM_PR_LABEL_BUILDS`?

Comment (pgarg66) [reply index.js]: Sounds good.

File: index.js [hunk 3]
```diff
@@ -139,7 +135,30 @@ async function triggerPullRequestCI(repoName, prNumber, commit) {
...
+  await prRemoveLabel(repoName, prNumber, label);
+}
+
+async function triggerLabelsOnPipeline(repoName, prNumber, commit) {
+  const customLabels = envconst.CUSTOM_PR_LABEL_BUILDS.split(',');
+
+  for (let index = 0; index < customLabels.length; ++index) {
+    let label = customLabels[index];
+    if (await prHasLabel(repoName, prNumber, label)) {
+      triggerPipeline(path.basename(repoName) + '-' + label, repoName, prNumber, commit, label);
```
Comment (mvines) [inline index.js]: `await triggerPipeline...`

**Comment by pgarg66** [comment]
@mvines looks like I don't have merge permissions for this. Could you merge it when you get a chance?

**Pull request merged**